Sources/Publish/API/AnyDateFormatter.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AnyDateFormatter.swift
+//  
+//
+//  Created by Dustin Pfannenstiel on 3/10/20.
+//  Copyright (c) Dustin Pfannenstiel, 2020
+//  MIT license, see LICENSE file for details
+//
+
+import Foundation
+
+/// A common interface to abstract common methods and properties present
+/// in all date formatter types.
+public protocol AnyDateFormatter: Formatter {
+    /// The time zone for the receiver.
+    var timeZone: TimeZone! { get set }
+    /// The date format string used by the receiver.
+    var dateFormat: String! { get }
+
+    /// Returns a date representation of a given string interpreted using the receiver’s current settings.
+    /// - parameter string: The string to parse.
+    func date(from string: String) -> Date?
+    /// Returns a string representation of a given date formatted using the receiver’s current settings.
+    /// - parameter date: The date to format.
+    func string(from date: Date) -> String
+}
+
+/// Apply `AnyDateFormatter` to `DateFormatter`
+extension DateFormatter: AnyDateFormatter {}
+
+/// Apply `AnyDateFormatter` to `ISO8601DateFormatter`
+@available(OSX 10.12, *)
+extension ISO8601DateFormatter: AnyDateFormatter {
+    /// Since `ISO8601DateFormatter` does not have a variable format, return a descriptive string for reporting.
+    public var dateFormat: String! {
```

Comment:
I think a better solution would be to remove this from the protocol requirements, as recommended above.

Reply:
I had two reasons for including this.  Initially because in at least [one place](https://github.com/dpfannenstiel/Publish/blob/7cd7de754618a36af0c52fa33ae82077fa7e46d7/Sources/Publish/Internal/MarkdownMetadataDecoder.swift#L627) Publish reads a `dateFormat` out of a `AnyDateFormatter`.
The second reason is that from an API design perspective what `AnyDateFormatter` is doing is defining the minimum standard interface for a date formatter. A data formatter without read access to a date format didn't make sense to me.  If anything `ISO8601DateFormatter` represented a special case that needed to be handled. This way instead of having to special case date formatter every time we need the `dateFormat` we now know the property is available.